src/matlab2tikz.m

```diff
@@ -963,23 +884,107 @@ case guitypes()
 end
 % ==============================================================================
 function legendhandle = getAssociatedLegend(m2t, handle)
-% Check if the axis is referenced by a legend (only necessary for Octave)
+legendhandle = [];
+env = getEnvironment;
+switch env
+    case 'Octave'
+        % Make sure that m2t.legendHandles is a row vector.
+        for lhandle = m2t.legendHandles(:)'
+            ud = get(lhandle, 'UserData');
+            % Empty if no legend and multiple handles if plotyy
+            if ~isempty(ud) && any(handle == ud.handle)
+                legendhandle = lhandle;
+                break
+            end
+        end
+    case 'MATLAB'
+        legendhandle = legend(handle);
+end
+
+% NOTE: there is a BUG in HG1 and Octave. Setting the box off sets the
+% legend visibility off too. We assume the legend is visible if it has 
+% a visible child.
+isInvisibleHG2 = isHG2() && ~isVisible(legendhandle);
+isInvisibleHG1orOctave = (~isHG2() || strcmpi(env,'Octave')) &&...
+    ~isVisibleContainer(legendhandle);
+
+% Do not return the handle if legend is invisible
+if isInvisibleHG1orOctave || isInvisibleHG2;
     legendhandle = [];
-    switch getEnvironment
-        case 'Octave'
-            % Make sure that m2t.legendHandles is a row vector.
-            for lhandle = m2t.legendHandles(:)'
-                ud = get(lhandle, 'UserData');
-                if isVisibleContainer(lhandle) && any(handle == ud.handle)
-                    legendhandle = lhandle;
-                    break;
-                end
+end
+end
+% ==============================================================================
+function entries = getLegendEntries(m2t)
+entries = [];
+legendHandle = m2t.axesContainers{end}.LegendHandle;
+
+if isempty(legendHandle), return, end
+
+switch getEnvironment
+    case 'Octave'
+        % See set(hlegend, "deletefcn", {@deletelegend2, ca, [], [], t1, hplots}); in legend.m
+        delfun  = get(legendHandle,'deletefcn');
+        entries = delfun{6};
+        
+        % Bubble-up legend entry properties from child to hggroup root
+        % for guessable objects
+        guessable = {'specgraph.errorbarseries','matlab.graphics.chart.primitive.ErrorBar',...
```

Comment:
Will this cause problems, once we add additional guessable plot types in octave and forget to adapt this line?